src/erlcloud_sm.erl

```diff
@@ -0,0 +1,136 @@
+
+%%%------------------------------------------------------------------------------
+%%% Library initialization.
+%%%------------------------------------------------------------------------------
+
+-spec new(string(), string()) -> aws_config().
+new(AccessKeyID, SecretAccessKey) ->
+    #aws_config{
+        access_key_id = AccessKeyID,
+        secret_access_key = SecretAccessKey
+    }.
+
+
+-spec new(string(), string(), string()) -> aws_config().
+new(AccessKeyID, SecretAccessKey, Host) ->
+    #aws_config{
+        access_key_id = AccessKeyID,
+        secret_access_key = SecretAccessKey,
+        sm_host = Host
+    }.
+
+
+-spec new(string(), string(), string(), non_neg_integer()) -> aws_config().
+new(AccessKeyID, SecretAccessKey, Host, Port) ->
+    #aws_config{
+        access_key_id = AccessKeyID,
+        secret_access_key = SecretAccessKey,
+        sm_host = Host,
+        sm_port = Port
+    }.
+
+%%------------------------------------------------------------------------------
+%% GetSecretValue
+%%------------------------------------------------------------------------------
+%% @doc
+%% SM API:
+%% [https://docs.aws.amazon.com/secretsmanager/latest/apireference/API_GetSecretValue.html]
+%% @end
+%%------------------------------------------------------------------------------
+
+-spec get_secret_value(SecretId :: binary(), Opts :: get_secret_value_options()) -> return_type().
+get_secret_value(SecretId, Opts) ->
+    get_secret_value(SecretId, Opts, erlcloud_aws:default_config()).
+
+
+-spec get_secret_value(SecretId :: binary(), Opts :: get_secret_value_options(),
+        Config :: aws_config()) -> return_type().
+get_secret_value(SecretId, Opts, Config) ->
+    Json = lists:map(
+        fun({Version, Val}) ->
+            case Version of
+                version_id -> {<<"VersionId">>, Val};
+                version_stage -> {<<"VersionStage">>, Val};
+                _ -> {Version, Val}
+            end
+        end,
+        [{<<"SecretId">>, SecretId} | Opts]),
+    sm_request(Config, "secretsmanager.GetSecretValue", Json).
+
+%%%------------------------------------------------------------------------------
+%%% Internal Functions
+%%%------------------------------------------------------------------------------
+
+sm_request(Config, Operation, Body) ->
+    case erlcloud_aws:update_config(Config) of
+        {ok, Config1} ->
+            sm_request_no_update(Config1, Operation, Body);
+        {error, Reason} ->
+            {error, Reason}
+    end.
+
+
+sm_request_no_update(Config, Operation, Body) ->
+    Payload = jsx:encode(Body),
+    Headers = headers(Config, Operation, Payload),
+    Request = #aws_request{service = sm,
+        uri = uri(Config),
+        method = post,
+        request_headers = Headers,
+        request_body = Payload},
+    case erlcloud_aws:request_to_return(erlcloud_retry:request(Config, Request, fun sm_result_fun/1)) of
+        {ok, {_RespHeaders, <<>>}} -> {ok, []};
```

Comment:
I would personally suggest just plain `ok` for this return, if and when you make API calls that return empty bodies.

Reply:
I prefer the current implementation; I think it is more consistent to return `{ok, Body}` regardless of the Body's content